Lib/doctest.py

```diff
@@ -207,7 +207,10 @@ def _normalize_module(module, depth=2):
     elif isinstance(module, str):
         return __import__(module, globals(), locals(), ["*"])
     elif module is None:
-        return sys.modules[sys._getframe(depth).f_globals['__name__']]
+        try:
+            return sys.modules[sys._getcallingmodule(depth)]
```

Comment:
That could raise KeyError if the module is None.

Reply:
I think the old code could, too? I can beef up the handling here, but I have a feeling when I looked at it it seemed best to let the exception escape.